_drafts/practical_zephyr_05_west.md

```diff
@@ -0,0 +1,1246 @@
+
+In the first article of this series, we installed the correct SDK for the [nRF52840 Development Kit](https://www.nordicsemi.com/Products/Development-hardware/nrf52840-dk) using [Nordic's toolchain manager](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/installation/assistant.html#install-toolchain-manager). This installation contains not only the Zephyr SDK but also the host dependencies that I need for building Zephyr applications for nRF devices.
+
+Since both, the STM32 and the nRF52840, are ARM MCUs, the installed Zephyr SDK already also contains the toolchain for building for the [STM32 Nucleo-64 development board](https://www.st.com/en/evaluation-tools/nucleo-f411re.html). All I need to do is provide the required environment variables.
+
+For that, I could, e.g., use a _shell_ script similar to what we've seen in the first article of this series:
+
+`app/setup-sdk-nrf.sh`
+
+```sh
+#!/bin/sh
+
+ncs_install_dir="${ncs_install_dir:-/opt/nordic/ncs}"
+ncs_bin_version="${ncs_bin_version:-4ef6631da0}"
+
+paths=(
+    $ncs_install_dir/toolchains/$ncs_bin_version/bin
+    $ncs_install_dir/toolchains/$ncs_bin_version/opt/nanopb/generator-bin
+)
+
+# required if dependencies are not installed, see also
+# https://docs.zephyrproject.org/latest/develop/getting_started/index.html#install-dependencies
+# e.g., "Ninja"
+for entry in ${paths[@]}; do
+    export PATH=$PATH:$entry
+done
+
+# only export the paths to the SDK, no longer export the path to the zephyr installation.
+export ZEPHYR_TOOLCHAIN_VARIANT=zephyr
+export ZEPHYR_SDK_INSTALL_DIR=$ncs_install_dir/toolchains/$ncs_bin_version/opt/zephyr-sdk
+```
+
+> **Notice:** Zephyr provides [environment scripts](https://docs.zephyrproject.org/latest/develop/env_vars.html#zephyr-environment-scripts) including a `zephyr-env.sh`, which you can source in case you're using [Zephyr's official SDK](https://docs.zephyrproject.org/latest/develop/toolchains/zephyr_sdk.html).
+
+In this script, I'm extending my `$PATH` for the binaries that come with Nordic's toolchain installation. This includes `cmake`, `ninja`, and all other host tools used by Zephyr. In fact, this installation even contains `git` and `python`, but by appending to `$PATH`, my local installations have precedence over Nordic's binaries.
+
+Then, I'm setting the `ZEPHYR_TOOLCHAIN_VARIANT` to `zephyr` and point `ZEPHYR_SDK_INSTALL_DIR` to the full path of the toolchain installation. Since I'm still using my own virtual environment, I need to install all Python packages required by Zephyr. Instead of cloning Zephyr as documented in the [getting started guide](https://docs.zephyrproject.org/latest/develop/getting_started/index.html#install-dependencies), I can use the locally populated Zephyr installation:
```

Comment:
It's not clear to me what last sentence here means. I might leave it out instead, it's clear to me that the next step is to install the Python dependencies

Reply:
In the getting started guide they instruct you to use `west init ~/zephyrproject` to get the Zephyr sources and to then install all dependencies using `pip install -r ~/zephyrproject/zephyr/scripts/requirements.txt`. I didn't find this an adequate solution since any Workspace application anyhow has Zephyr as a project.

Would you have a suggestion how to phrase this instead?